packages/fxa-auth-server/lib/routes/totp.js

```diff
@@ -702,5 +712,202 @@ module.exports = (
         }
       },
     },
+    {
+      /**
+       * This endpoint explicitly does not modify the recoveryCodes
+       * in the same way that `/totp/create` does. We want to allow
+       * changing the recovery method without changing the backup codes.
+       */
+      method: 'POST',
+      path: '/totp/replace/start',
+      options: {
+        ...TOTP_DOCS.TOTP_REPLACE_START_POST,
+        auth: {
+          strategy: 'sessionToken',
+          payload: 'required',
+        },
+        validate: {
+          payload: isA.object({
+            metricsContext: METRICS_CONTEXT_SCHEMA,
+          }),
+        },
+        response: {
+          schema: isA.object({
+            qrCodeUrl: isA.string().required(),
+            secret: isA.string().required(),
+          }),
+        },
+      },
+      handler: async function (request) {
+        log.begin('totp.replace.create', request);
+
+        const sessionToken = request.auth.credentials;
+        const uid = sessionToken.uid;
+
+        await customs.checkAuthenticated(
+          request,
+          uid,
+          sessionToken.email,
+          'totpCreate'
+        );
+
+        if (sessionToken.tokenVerificationId) {
+          throw errors.unverifiedSession();
+        }
+
+        // the opposite of `/totp/create` this requires that the user already has
+        // a verified TOTP token to be replaced.
+        const hasEnabledToken = await otpUtils.hasTotpToken({ uid });
+        if (!hasEnabledToken) {
+          throw errors.totpTokenDoesNotExist();
+        }
+
+        // Default options for TOTP
+        const otpOptions = {
+          encoding: 'hex',
+          step: config.step,
+          window: config.window,
+        };
+
+        const authenticator = new otplib.authenticator.Authenticator();
+        authenticator.options = Object.assign(
+          {},
+          otplib.authenticator.options,
+          otpOptions
+        );
+
+        const secret = authenticator.generateSecret();
+        await authServerCacheRedis.set(
+          toRedisTotpSecretKey(uid),
+          secret,
+          'EX',
+          TOTP_SECRET_REDIS_TTL
+        );
+
+        log.info('totpToken.replace.created', { uid });
+        await request.emitMetricsEvent('totpToken.replace.created', { uid });
+
+        const otpauth = authenticator.keyuri(
+          sessionToken.email,
+          service,
+          secret
+        );
+
+        const qrCodeUrl = await qrcode.toDataURL(otpauth, qrCodeOptions);
+
+        return {
+          qrCodeUrl,
+          secret,
+        };
+      },
+    },
+    {
+      method: 'POST',
+      path: '/totp/replace/confirm',
```

Comment:
Just so you have context on `confirm` vs `verify`, we used to use the word "verify" everywhere and a couple of years ago we changed all the copy in the front-end to "confirm". Looks like we've got `/confirm` for recovery phone stuff so I don't have a preference on the endpoint name here.